src/js/core/widget/core/MainTab.js

```diff
@@ -0,0 +1,237 @@
+					/**
+					 * Set Main Tab widget as active (visible)
+					 * @style ui-main-tab-active
+					 * @member ns.widget.core.MainTab
+					 */
+					ACTIVE: "ui-active",
+					/**
+					 * Set Tab as active
+					 * @style ui-tab-active
+					 * @member ns.widget.core.MainTab
+					 */
+					ACTIVE_TAB: "ui-tab-active"
+				},
+				MainTab = function () {
+					this._ui = {};
+				},
+				prototype = new BaseWidget();
+
+			MainTab.classes = classes;
+			MainTab.prototype = prototype;
+
+			/**
+			 * Build MainTab component
+			 * @method _build
+			 * @param {HTMLElement} element
+			 * @return {HTMLElement} Returns built element
+			 * @member ns.widget.core.MainTab
+			 * @protected
+			 */
+			prototype._build = function (element) {
+				element.classList.add(classes.MAIN_TAB);
+
+				return element;
+			};
+
+			/**
+			 * Init MainTab component
+			 * @method _init
+			 * @param {HTMLElement} element
+			 * @return {HTMLElement} Returns built element
+			 * @member ns.widget.core.MainTab
+			 * @protected
+			 */
+			prototype._init = function (element) {
+				var self = this,
+					pageContainer = null;
+
+				self._ui.parentPage = utilSelectors.getClosestBySelector(element, ".ui-page");
+
+				self._ui.links = [].slice.call(element.querySelectorAll("li > a"));
+				self._ui.links.forEach(function (link) {
+					var wrap = document.createElement("span"),
+						content = link.textContent.trim();
+
+					wrap.textContent = content;
+					link.textContent = "";
+					link.appendChild(wrap);
+
+					ns.widget.Marquee(wrap, {
+						iteration: 1,
+						delay: 0,
+						marqueeStyle: "endToEnd",
+						autoRun: false
+					});
+					link.setAttribute("data-rel", "maintab");
+				});
+
+				element.classList.add(classes.ACTIVE);
+
+				// change parent to pageContainer instead of page
+				pageContainer = utilSelectors.getClosestBySelector(element, ".ui-page-container");
+				pageContainer.appendChild(element);
+
+				return element;
+			};
+
+			/**
+			 * Handle events
+			 * @param {Event} event
+			 * @protected
+			 * @member ns.widget.core.MainTab
+			 */
+			prototype.handleEvent = function (event) {
+				var self = this;
+
+				switch (event.type) {
+					case "pagebeforeshow":
+						self._onPageBeforeShow(event);
+						break;
+				}
+			};
+
+			prototype._bindEvents = function () {
+				document.addEventListener("pagebeforeshow", this, true);
+			};
+
+			prototype._unbindEvents = function () {
+				document.removeEventListener("pagebeforeshow", this, true);
+			};
+
+			prototype._onPageBeforeShow = function (event) {
+				var self = this,
+					target = event.target,
+					element = self.element,
+					links = self._ui.links,
+					router,
+					activeLink,
+					url,
+					uri;
+
+				if (self._ui.parentPage === event.target) {
+					// try to find active tab
+					activeLink = links.filter(function (link) {
+						return link.classList.contains(classes.ACTIVE_TAB);
+					})[0] || element.querySelector("a");
```

Comment:
Isn't there a possibility that activeLink could be null?

Reply:
This is possible if developer didn't add any tab to MainTab, maybe it is actually better to protect it.

Reply:
Protection has been added.